docs/java-run-your-first-app.md

```diff
@@ -0,0 +1,160 @@
+<!--SNIPEND-->
+
+Notice that the Worker will listen to the same Task Queue that the Workflow and Activity tasks are sent to. This is called "Task routing", and is a built-in mechanism for load balancing.
+
+<!--SNIPSTART money-transfer-project-template-java-shared-constants-->
+<!--SNIPEND-->
+
+Let's start the Worker. From within IntelliJ run the TransferMoneyWorker class, or from the project root, run the following command:
+
+```
+./gradlew startWorker
+```
+
+When the Worker starts it begins polling the Task Queue. The first Task it finds is the Workflow Task, which it executes. Executing the Workflow causes the Activity Tasks to be sent to the Task Queue as well. The Worker then grabs each of the Activity Tasks in their respective order from the Task Queue and executes each of the corresponding Activities.
+
+<img class="docs-image-centered" src={require('../static/img/docs/confetti.png').default} />
+
+**Congratulations**, you have run a Temporal Workflow application!
+
+## Failure simulation &nbsp;&nbsp;![](/img/docs/warning.png)
+
+By running the application as is, at minimum you got a taste of one of Temporal's amazing value propositions: visibility into the Workflow and the status of the Workers executing the code. Let's explore another key value proposition, maintaining the state of a Workflow, by simulating some failures. Make sure your Worker is stopped before proceeding.
+
+### Server crash
+
+In most modern application environments, to accommodate a host failure, leader election processes are applied to multiple application instances which are backed by databases that maintain the application state and so on. With Temporal, even if the Temporal server is down, the state of the Workflow is preserved. You can easily test this by following these steps (again, make sure your Worker is stopped so your Workflow doesn't finish):
+
+1. Start the Workflow again.
+2. Verify the Workflow is running in the UI.
+3. Shut down the Temporal server by either using 'Ctrl c' or via the Docker dashboard.
+4. After it has stopped, restart the Temporal server and visit the UI.
+
+The Workflow is still there!
+
+### Activity error
+
+Next let's simulate a bug in the Activity method. Inside your project find the AccountActivityImpl.java file and uncomment the line that throws an exception in the deposit method:
+
+<!--SNIPSTART money-transfer-project-template-java-activity-implementation-->
+<!--SNIPEND-->
+
+Save it and run the Worker (the Workflow should still be running, start it if it isn't). The Worker executes the Workflow, and completes the `withdraw()` Activity. But then the exception is thrown when it attempts the `deposit()` Activity. Notice how the Worker keeps retrying the Activity? Visit the [UI](localhost:8088) and click on the RunId of the Workflow. You will see the pending Activity listed there with details such as its state, the number of times it has been attempted, and the next scheduled attempt.
```

Comment:
```suggestion
After uncommenting the exception save your changes and run the Worker (the Workflow should still be running, start it if it isn't). The Worker executes the Workflow, and completes the `withdraw()` Activity method. But then an exception is thrown when it reaches the `deposit()` Activity method. Notice how the Worker keeps retrying the deposit Activity? Visit the [UI](localhost:8088) and click on the RunId of the Workflow. You will see the pending Activity listed there with details such as its state, the number of times it has been attempted, and the next scheduled attempt.
```

"The workflow should be running, start if isnt" that suggests we could have better instructions


In general I felt more lost at this part than the rest of the tutorial.

Reply:
I refined this a bit - LMK.